packages/network/src/createTxQueue.ts

```diff
@@ -227,14 +230,31 @@ export function createTxQueue<C extends Contracts>(
           error &&
           (("code" in error && error.code === "NONCE_EXPIRED") ||
             JSON.stringify(error).includes("transaction already imported"));
+        const incorrectGasEstimation =
+          error &&
+          (JSON.stringify(error).includes("gas required exceeds allowance") ||
+            JSON.stringify(error).includes("intrinsic gas too low"));
```

Comment:
do these errors occur in different situations? What are the situations in which each one occurs?

Reply:
My main concern here is with handling situations where we need to increase the nonce ([tx was included in a block but failed](https://etherscan.io/tx/0xbf7146c7799b8c1e486260726ad5f383f1e98ee9c726b6a76a7e4d1e57f7c298)) vs those where we don't (tx was declined by ethers or maybe the RPC?). Is there any information in the error that let's us tell for sure if the tx was included in a block? Maybe an abi encoded "Out of Gas" exception or something?

Reply:
if it's included in a block you get a tx hash back and no error. to check error in a tx in a block you need to fetch the tx details from the tx hash.

Reply:
> if it's included in a block you get a tx hash back and no error. to check error in a tx in a block you need to fetch the tx details from the tx hash.

now that i know this, i think this whole PR is misguided. by fetching the tx details we are starting another round trip to the chain. at that point we might as well submit again via other means. i think this whole retrying issue will have to be moved closer to the application layer.